app/main.py

```diff
@@ -1,13 +1,47 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8",
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [
+            {
+                "line": error["line_number"],
+                "column": error["column_number"],
+                "message": error["text"],
+                "name": error["code"],
+                "source": "flake8",
+            }
+            for error in errors
+        ],
+        "path": file_path,
+        "status": "passed" if len(errors) == 0 else "failed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        {
+            "errors": [] if len(linter_report[f"{report}"]) == 0 else [
+                {
+                    "line": error["line_number"],
+                    "column": error["column_number"],
+                    "message": error["text"],
+                    "name": error["code"],
+                    "source": "flake8",
+                }
+                for error in linter_report[f"{report}"]
+            ],
+            "path": report,
+            "status": "passed" if
+            len(linter_report[f"{report}"]) == 0
+            else "failed"
```

Comment:
you can use .items() method of dict that returns (key, value) tuples, thus you can iterate through keys and values at the same time instead of fetching value for each key every time